src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformConfigForm.tsx

```diff
@@ -0,0 +1,431 @@
+/*
+ * Copyright 2021 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Shield 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+import React, { useEffect } from 'react'
+import {
+  Button,
+  Color,
+  Formik,
+  Layout,
+  Text,
+  Heading,
+  Card,
+  Icon,
+  ButtonVariation,
+  ButtonSize,
+  FormInput,
+  SelectOption,
+  getMultiTypeFromValue,
+  MultiTypeInputType,
+  StepProps
+} from '@wings-software/uicore'
+import cx from 'classnames'
+import * as Yup from 'yup'
+import { useParams } from 'react-router-dom'
+import { Form } from 'formik'
+import { useVariablesExpression } from '@pipeline/components/PipelineStudio/PiplineHooks/useVariablesExpression'
+import { useStrings } from 'framework/strings'
+import { FormMultiTypeConnectorField } from '@connectors/components/ConnectorReferenceField/FormMultiTypeConnectorField'
+import { ConfigureOptions } from '@common/components/ConfigureOptions/ConfigureOptions'
+import {
+  AllowedTypes,
+  tfVarIcons,
+  ConnectorMap,
+  ConnectorLabelMap,
+  ConnectorTypes,
+  formInputNames,
+  formikOnChangeNames,
+  stepTwoValidationSchema
+} from './TerraformConfigFormHelper'
+
+import type { Connector } from '../TerraformInterfaces'
+
+import css from './TerraformConfigForm.module.scss'
+import stepCss from '@pipeline/components/PipelineSteps/Steps/Steps.module.scss'
+
+interface TerraformConfigStepOneProps {
+  data: any
+  isReadonly: boolean
+  allowableTypes: MultiTypeInputType[]
+  isEditMode: boolean
+  selectedConnector: string
+  setConnectorView: (val: boolean) => void
+  setSelectedConnector: (val: ConnectorTypes) => void
+  isTerraformPlan?: boolean
+}
+
+export const TerraformConfigStepOne: React.FC<StepProps<any> & TerraformConfigStepOneProps> = ({
+  data,
+  isReadonly,
+  allowableTypes,
+  nextStep,
+  setConnectorView,
+  selectedConnector,
+  setSelectedConnector,
+  isTerraformPlan = false
+}) => {
+  const { getString } = useStrings()
+  const { expressions } = useVariablesExpression()
+  const { accountId, projectIdentifier, orgIdentifier } = useParams<{
+    projectIdentifier: string
+    orgIdentifier: string
+    accountId: string
+  }>()
+
+  useEffect(() => {
+    const selectedStore = isTerraformPlan
+      ? data?.spec?.configuration?.configFiles?.store?.type
+      : data?.spec?.configuration?.spec?.configFiles?.store?.type
+    setSelectedConnector(selectedStore)
+  }, [])
```

Comment:
Codacy has a fix for the issue: [React Hook useEffect has missing dependencies: 'data?.spec?.configuration?.configFiles?.store?.type', 'data?.spec?.configuration?.spec?.configFiles?.store?.type', 'isTerraformPlan', and 'setSelectedConnector'. Either include them or remove the dependency array. If 'setSelectedConnector' changes too often, find the parent component that defines it and wrap that definition in useCallback.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232256)
```suggestion
  }, [data?.spec?.configuration?.configFiles?.store?.type, data?.spec?.configuration?.spec?.configFiles?.store?.type, isTerraformPlan, setSelectedConnector])
```